rest/oidc_api.go

```diff
@@ -228,15 +228,18 @@ func (h *handler) getOIDCProvider(providerName string) (*auth.OIDCProvider, erro
 
 // Builds the OIDC callback based on the current request and database. Used during OIDC Client lazy initialization.  Needs to pass
 // in dbName, as it's not necessarily initialized on the request yet.
-func (h *handler) getOIDCCallbackURL() string {
+func (h *handler) getOIDCCallbackURL(providerName string, isDefault bool) string {
+	callbackPath := "_oidc_callback"
 	scheme := "http"
 	if h.rq.TLS != nil {
 		scheme = "https"
 	}
 	if dbName := h.PathVar("db"); dbName == "" {
 		base.Warnf("Can't calculate OIDC callback URL without DB in path.")
 		return ""
+	} else if !isDefault && providerName != "" {
+		return fmt.Sprintf("%s://%s/%s/%s?%s=%s", scheme, h.rq.Host, dbName, callbackPath, auth.OIDCAuthProvider, providerName)
```

Comment:
Should we still be using the same underlying `SetURLQueryParam` instead of doing the same logic in a different way?

Reply:
Make sense; We can make use of SetURLQueryParam here. Initially, I didn't realize any value in parsing the URL string here since we know that there is no parameter in the URL at this point. But yes, it is reasonable to validate created callback URL before setting the provider.